pkg/collector/check/check.go

```diff
@@ -39,4 +39,27 @@ type Check interface {
 	ConfigSource() string
 	// IsTelemetryEnabled returns if telemetry is enabled for this check
 	IsTelemetryEnabled() bool
+	// InitConfig returns the init_config configuration of the check
+	InitConfig() string
```

Comment:
Why is this a `string`? I would have expected it to be a struct with the configuration?

Reply:
> It's up to each checks to load it's YAML configuration as only the check knows the structure (field name, nested field, types, ...). What we want here is the YAML configuration not the actual field.

I guess what confuses me is that I would expect that the YAML configuration is already loaded when you are able to call `InitConfig`. But it's probably a big change, so no point in discussing it here.

What would help (and may avoid some programming errors) is to have a type alias like `type YAMLConfig string`